### PR TITLE
[custom certs] Updates charts to support custom certificate

### PR DESCRIPTION
**Pull request by GoelDeepak**
This patch is to improve the flow around using custom certificate with Konvoy. 

We recently added a feature in Konvoy in which it could create extra Kubernetes objects if an operator places them in `extras/kubernetes` directory.  For a custom certificate, we will use this feature of Konvoy. The expectation is that the operator would create a secret from the custom certificate files and then specify the secret name `caSecretName` as part of the values for each addon in `cluster.yaml` 

This PR changes the chart of 4 addons to use `caSecretName` if it is specified 

Additionally, this PR also displays the custom certificate on the `/token` output. Prior to this PR, it use to always display Kubernetes CA even if an operator is using custom certificate

### CODE REVIEW

**Comment by GoelDeepak** [comment]
I have manually carried out two tests:
1. Functional testing - Verified everything is working as expected
2. Upgrade testing - Verified that charts are upgradable

File: staging/dex-k8s-authenticator/values.yaml [hunk 1]
```diff
@@ -73,6 +73,8 @@ resources: {}
 
 caCerts:
   enabled: false
```
Comment (hectorj2f) [inline staging/dex-k8s-authenticator/values.yaml]: `enabled: false ?`

File: staging/traefik/values.yaml [hunk 2]
```diff
@@ -136,7 +136,7 @@ ssl:
   # tlsMinVersion: VersionTLS12
   # https://docs.traefik.io/configuration/entrypoints/#strict-sni-checking
   # sniStrict: false
-  customCert: false
```
Comment (hectorj2f) [inline staging/traefik/values.yaml]: what about doing something like https://github.com/mesosphere/charts/pull/369/files#diff-84491f379c2eed7f94384116a9e49e45R74 with a boolean and a secret name ?

File: staging/dex-k8s-authenticator/html-templates/linux-mac-common.html [hunk 3]
```diff
@@ -90,8 +90,8 @@ <h3>Run configuration commands</h3>
       <img class="clippy" width="13" src="{{ .Web_Path_Prefix }}static/clippy.svg" alt="">
     </button>
     <pre><code class="hljs">kubectl config set-context {{ .Username }}-{{ .ClusterName }} \
-    --cluster={{ .ClusterName }} \
-    --user={{ .Username}}-{{.ClusterName }}</code></pre>
+    --cluster={{ .Username }}-{{ .ClusterName }} \
+    --user={{ .Username }}-{{.ClusterName }}</code></pre>
```
Comment (hectorj2f) [inline staging/dex-k8s-authenticator/html-templates/linux-mac-common.html]: `-{{ .ClusterName }}`

File: staging/dex-k8s-authenticator/html-templates/linux-mac-common.html [hunk 4]
```diff
@@ -8,7 +8,7 @@ <h3>Copy IDP CA Certificate From URL</h3>
       <button class="btn" style="float:right" data-clipboard-snippet="">
         <img class="clippy" width="13" src="{{ .Web_Path_Prefix }}static/clippy.svg" alt=""/>
       </button>
-      <pre><code>curl --create-dirs -s {{ .IDPCaURI }} -o ${HOME}/.kube/certs/{{ .ClusterName }}/idp-ca.crt</code></pre>
+      <pre><code>curl --create-dirs -s {{ .IDPCaURI }} -o ${HOME}/.kube/certs/{{ .Username }}-{{ .ClusterName }}/idp-ca.crt</code></pre>
```
Comment (hectorj2f) [inline staging/dex-k8s-authenticator/html-templates/linux-mac-common.html]: question: where is the Username coming from ?

Comment (hectorj2f) [reply staging/dex-k8s-authenticator/html-templates/linux-mac-common.html]: Would that work when attaching existing Kubernetes clusters like GKE and deploying addons there ?

File: staging/dex-k8s-authenticator/values.yaml (hunk 1 above)
Comment (GoelDeepak) [reply staging/dex-k8s-authenticator/values.yaml]: yeah, the default is false. It is enabled by the operator if they want to use custom certs

File: staging/traefik/values.yaml (hunk 2 above)
Comment (GoelDeepak) [reply staging/traefik/values.yaml]: Sorry, didn't understand your comment. This variable was introduced by us and this patch is taking it out

**Comment by alejandroEsc** [comment]
Going to merge as we need this for tuesday and we need to test changes against these asap.

File: staging/traefik/values.yaml (hunk 2 above)
Comment (hectorj2f) [reply staging/traefik/values.yaml]: I meant, why don't you respect the same structure and have a complex object with a boolean and a secretName as done in https://github.com/mesosphere/charts/pull/369/files#diff-84491f379c2eed7f94384116a9e49e45R74

File: staging/dex-k8s-authenticator/html-templates/linux-mac-common.html (hunk 3 above)
Comment (hectorj2f) [reply staging/dex-k8s-authenticator/html-templates/linux-mac-common.html]: @alejandroEsc ^

File: staging/traefik/values.yaml (hunk 2 above)
Comment (GoelDeepak) [reply staging/traefik/values.yaml]: What is the benefit of that structure?

**Pull request merged**